aws/table_aws_cloudtrail_lake_query.go

```diff
@@ -0,0 +1,199 @@
+			},
+			{
+				Name:        "bytes_scanned",
+				Description: "Gets metadata about a query, including the number of events that were matched, the total number of events scanned, the query run time in milliseconds, and the query's creation time.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudtrailLakeQuery,
+				Transform:   transform.FromField("QueryStatistics.BytesScanned"),
+			},
+			{
+				Name:        "events_matched",
+				Description: "The number of events that matched a query.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudtrailLakeQuery,
+				Transform:   transform.FromField("QueryStatistics.EventsMatched"),
+			},
+			{
+				Name:        "events_scanned",
+				Description: "The number of events that the query scanned in the event data store.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudtrailLakeQuery,
+				Transform:   transform.FromField("QueryStatistics.EventsScanned"),
+			},
+			{
+				Name:        "execution_time_in_millis",
+				Description: "The query's run time, in milliseconds.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudtrailLakeQuery,
+				Transform:   transform.FromField("QueryStatistics.ExecutionTimeInMillis"),
+			},
+			{
+				Name:        "query_string",
+				Description: "The SQL code of a query.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getCloudtrailLakeQuery,
+			},
+
+			// steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("QueryId"),
+			},
+		}),
+	}
+}
+
+// We need custom structure
+// 1. EventDataStoreArn is required for joing the queries
+// 2. list/get call do not provide the EventDataStoreArn
+// 3. Both list/get API call return different stracture with dfferent data
+type QueryInfo struct {
+	EventDataStoreArn *string
+	CreationTime     *time.Time
+	QueryId          *string
+	QueryStatus      types.QueryStatus
+	QueryString      *string
+	QueryStatistics  *types.QueryStatisticsForDescribeQuery
+}
+
+//// LIST FUNCTION
+
+func listCloudtrailLakeQueries(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	eventDataStore := h.Item.(types.EventDataStore)
+
+	if d.KeyColumnQualString("event_data_store_arn") != "" {
+		if d.KeyColumnQualString("event_data_store_arn") != *eventDataStore.EventDataStoreArn {
+			return nil, nil
+		}
+	}
+
+	// Get client
+	svc, err := CloudTrailClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Info("aws_cloudtrail_lake_query.listCloudtrailLakeQueries", "client_error", err)
+		return nil, err
+	}
+
+	// Limiting the results
+	maxLimit := int32(500)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
+				maxLimit = 1
+			} else {
+				maxLimit = limit
+			}
+		}
+	}
+
+	input := &cloudtrail.ListQueriesInput{
```

Comment:
From https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudtrail#Client.ListQueries, it seems like only the queries from the past 7 days are returned. How would we expect users to know this and how would we expect them to interact with the table? Do they need to specify `creation_time` in a query then?

Reply:
There is no data available in our console, which is older than seven days, so we could not test this scenario.
But if we pass the `creation_time` value in the where clause, it returns the result instead.

Reply:
I believe the table doc should mention the default behaviour (the table returns queries from the past 7 days), and we should ask users to specify a creation date if they want results from further back. We should also add an example showing how to specify `creation_time`.

Reply:
Agree, using the `StartTime` and `EndTime` attributes is the right direction. How many more days do we need to wait from today, December 6, in order to have queries older than 7 days?